.github/workflows/build-and-push-image.yaml

```diff
@@ -0,0 +1,40 @@
+name: Create and publish a Docker image
+
+on:
+  push:
+    branches: ['main', 'master']
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
```

Comment:
what’s the rationale here for using a specific committee vs. `v2`?

Reply:
It's a security feature.

A git tag can be overwritten at any time, which means an attacker could replace the current `v2` by malicious code which would then be used in our workflow.

As long as we assume that hash collisions are unlikely / impossible, this cannot happen when tagging a specific commit.